### PR TITLE
Adding additional apis for DateTimeAssert

### PR DESCRIPTION
**Pull request by johnkillmer**
Adding apis for 
hasYear, 
hasMonthOfYear, 
hasDayOfMonth, 
hasHourOfDay, 
hasMinuteOfHour, 
hasSecondsOfMinute, 
hasMillisOfSecond

### CODE REVIEW

File: src/main/java/org/assertj/jodatime/api/DateTimeAssert.java [hunk 1]
```diff
@@ -51,6 +68,139 @@ protected DateTime getActual() {
...
+   * Example :
+   * <pre><code class='java'> assertThat(new DateTime(&quot;2000-01-01&quot;)).hasYear(1);</code></pre>
+   *
+   * @param other the given year.
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code DateTime} is {@code null}.
+   * @throws IllegalArgumentException if other {@code DateTime} is {@code null}.
+   * @throws AssertionError if the year of the actual {@code DateTime} is not equal to the given year.
+   */
+  public DateTimeAssert hasYear(Integer other){
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/jodatime/api/DateTimeAssert.java]: Any reason for using an `Integer` in place of an `int` ?  that would spare us to check for a null parameter.

Comment (johnkillmer) [reply src/main/java/org/assertj/jodatime/api/DateTimeAssert.java]: No reason, just didn't think about that.  I'll change it to an int. I like that suggestion.

**Comment by johnkillmer** [comment]
PR should be updated. Used int instead of Integer. Removed the unneeded null check. Fix the Asssertion error messages. Had the expected and actual mixed up.

File: src/main/java/org/assertj/jodatime/api/DateTimeAssert.java [hunk 2]
```diff
@@ -51,6 +59,125 @@ protected DateTime getActual() {
...
+   * <p>
+   * Example :
+   * <pre><code class='java'> assertThat(new DateTime(&quot;2000-01-01&quot;)).hasYear(1);</code></pre>
+   *
+   * @param other the given year.
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code DateTime} is {@code null}.
+   * @throws AssertionError if the year of the actual {@code DateTime} is not equal to the given year.
+   */
+  public DateTimeAssert hasYear(int other){
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/jodatime/api/DateTimeAssert.java]: did not think about it at my first review but can we name `other` parameters according to their meaning, here it should be `year` or `expectedYear` (pick the one you like more).

**Comment by joel-costigliola** [comment]
@johnkillmer looks good thanks.

I did not think about it at my first review but `other` parameters should be renamed according to their meaning, for example in `hasYear` it should be `year` or `expectedYear` (pick the one you like more).
Obviously this should be done for the other assertions.

File: src/main/java/org/assertj/jodatime/api/DateTimeAssert.java [hunk 3]
```diff
@@ -599,6 +726,15 @@ private void checkIsNotNullAndNotEmpty(Object[] values) {
     if (values.length == 0) throw new IllegalArgumentException("The given DateTime array should not be empty");
   }
 
+
+  /**
+   *
+   * @param other
+   */
+  private static void assertParameterIsNotNull(Integer integer, String exceptionMessage) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/jodatime/api/DateTimeAssert.java]: is that still used since we pass `int` now ?

**Comment by johnkillmer** [comment]
@joel-costigliola   I agree, those are much better names.  Removed that unused method as well. PR is updated. 

Thanks for the feedback

**Comment by johnkillmer** [comment]
@joel-costigliola Do you have an idea when you might release the next version of assertj-doda-time?

**Comment by joel-costigliola** [comment]
Once the existing issues are addressed, likely in a few weeks.

**Comment by joel-costigliola** [comment]
Integrated thanks @johnkillmer!

**Pull request closed without merging**